jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -549,12 +549,20 @@ private void checkClassAccess(Class<?> targetClass) throws IllegalAccessExceptio
 			throw new IllegalAccessException(com.ibm.oti.util.Msg.getString("K0680", accessClass.getName(), targetClass.getName()));  //$NON-NLS-1$
 		}
 		
-		private void checkSpecialAccess(Class<?> callerClass) throws IllegalAccessException {
+		private void checkSpecialAccess(Class<?> declaringClass, Class<?> callerClass) throws IllegalAccessException {
 			if (INTERNAL_PRIVILEGED == accessMode) {
 				// Full access for use by MH implementation.
 				return;
 			}
-			if (isWeakenedLookup() || accessClass != callerClass) {
+			/* Given that findSpecial emulates the behavior of invokespecial,
+			 * calling interface methods of declaringClass should be allowed as long as 
+			 * declaringClass is an interface and callerClass can be assigned to it when
+			 * callerClass is not the same as the lookup class.
+			 */
+			if (isWeakenedLookup() 
+			|| ((accessClass != callerClass) 
+			&& !((null != declaringClass) && declaringClass.isInterface() && declaringClass.isAssignableFrom(callerClass)))
```

Comment:
When can `declaringClass` be null?  I'm not sure I saw a case that allowed that.

Reply:
Already removed as unnecessary.